src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -403,8 +403,7 @@ SetTaskQueryIfShouldLazyDeparse(Task *task, Query *query)
 		return;
 	}
 
-	SetTaskQueryString(task, AnnotateQuery(DeparseTaskQuery(task, query),
-										   task->partitionColumn, task->colocationId));
+	SetTaskQueryString(task, DeparseTaskQuery(task, query));
```

Comment:
we sometimes cache the plan for local execution and reuse it for different distribution key values. Would we still be parsing the annotation in that case?

Reply:
My intention was to handle cached queries properly. However, upon discussing it with Onder, we observed that this PR doesn't do so. During my testing, I set citus.log_remote_commands to on, which prevented query caching. As our monitoring logic relies on query annotation, we need an actual query to track metrics for cached queries. Therefore, I'm currently exploring alternative options. Do you have any suggestions?

Reply:
I think you can just use the partitionKeyValue in CitusEndScan